File: java-playground/src/test/java/org/gonza/javaplayground/lotto/domain/utils/RandomNumberGeneratorTest.java
package org.gonza.javaplayground.lotto.domain.utils;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class RandomNumberGeneratorTest {

    private RandomNumberGenerator sut;

    @BeforeEach
    public void setUp() {
        sut = new RandomNumberGenerator();
    }

    @Test
    public void should_generate_number_between_min_max_and_specific_size() {
        Integer size = 6;
        Integer min = 1;
        Integer max = 6;

        List<Integer> numbers = sut.generate(size, min, max);

        Integer minNumber = numbers.stream().min(Integer::compareTo).get();
        Integer maxNumber = numbers.stream().max(Integer::compareTo).get();

        assertEquals(min, minNumber);
        assertEquals(max, maxNumber);
        assertEquals(size, numbers.size());
    }

    @Test
    public void should_throw_if_min_is_larger_than_max() {
        Integer size = 6;
        Integer min = 6;
        Integer max = 1;

        assertThrows(IllegalArgumentException.class, () -> sut.generate(size, min, max));
    }
}

File: kotlin-playground/src/test/kotlin/org/gonza/kotlinplayground/service/RandomNumberGeneratorTest.kt
package org.gonza.kotlinplayground.service

import org.gonza.kotlinplayground.domain.lotto.NumberRange
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class TestNumberRange : NumberRange {
    override fun getMin(): Int = 0

    override fun getMax(): Int = 0
}

class RandomNumberGeneratorTest {
    @Test
    fun `최대값과 최소값을 넘지 않는 값을 생성할 수 있다`() {
        val testNumberRange = TestNumberRange()
        val randomNumberGenerator = RandomNumberGenerator(testNumberRange)

        val result =
            listOf(
                randomNumberGenerator.get(),
                randomNumberGenerator.get(),
                randomNumberGenerator.get(),
                randomNumberGenerator.get(),
                randomNumberGenerator.get(),
                randomNumberGenerator.get(),
                randomNumberGenerator.get(),
            )

        result.forEach {
            assertTrue(it >= testNumberRange.getMin())
            assertTrue(it <= testNumberRange.getMax())
        }
    }
}
